mbed_build/_internal/config/config.py

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Build configuration abstraction layer."""
+from dataclasses import dataclass, field
+from typing import List
+
+from mbed_build._internal.config.config_layer import ConfigLayer
+
+
+@dataclass
+class Config:
+    """Represents a build configuration."""
+
+    settings: dict = field(default_factory=dict)
+
+    @classmethod
+    def from_layers(cls, layers: List[ConfigLayer]) -> "Config":
+        """Create configuration from layers."""
```

Comment:
```suggestion
        """Create configuration from `mbed_build.config_layer.ConfigLayer`."""
```

Reply:
We don't generate docs for the `_internal` modules, so I'm not sure what the value of this is. Also, the type hint indicates what the argument is. Thoughts?

Reply:
That is true. I had forgotten that only public modules are documented. In that case I think you are right and type hints are enough.